zealot/src/fetcher/fetcher.ts

```diff
@@ -24,6 +24,10 @@ export function createFetcher(host: string) {
     async stream(args: FetchArgs) {
       const {path, method, body, signal} = args
       const resp = await fetch(url(host, path), {method, body, signal})
+      if (!resp.ok) {
+        const content = await parseContentType(resp)
+        return Promise.reject(createError(content))
+      }
```

Comment:
This check existed in the next function `createIterator` (https://github.com/brimsec/brim/blob/master/zealot/src/fetcher/iterator.ts#L12), but it does feel like it belongs where you put it here. I like it. Would you remove it from the `createIterator` function and copy the `isObject(error)` check?

Reply:
Sure

Reply:
Looks like the `createError` method already does the check so I don't think we need to include it.